FogCreek/Glitch-Community
# Jet socks: modularize new project pop

## Links
* Remix link: https://jet-socks.glitch.me/

## GIF/Screenshots
![self-adjusting-popover](https://user-images.githubusercontent.com/938722/57810155-289f2080-7735-11e9-8c0f-b1d4986560e0.gif)

## Changes
* create Popover components
* create ResultsList component (possibly only used in popovers, but not really popover-specific)
* incorporate new-project-pop into Header component
* use the v1 API for fetching new projects

## How To Test
* visit https://jet-socks.glitch.me/
* clicking New Project should open a menu aligned to the right of the button
* if you make the window narrow enough that it would go off the edge of the screen, it should adjust to fit
* log in
* the New Project menu should now align to the left of the button

## Feedback I'm looking for
* I think the right approach for this conversion is to work on fully converting each popover one by one, copying styles and components as i go, and then delete all of the legacy styles/presenters when they're all complete; I think this will be easier than trying to convert a single component across multiple popovers. Does this seem sensible?
* ~I left new-project-pop in presenters/pop-overs because I would categorize it as something that's specific to our app, rather than something that someone else would use to build their own app; its more like a very small page than a very complicated component. That said, we have a ton of stuff in components that is also more content-ish than component-ish (e.g. the header & footer). Any thoughts on how we should organize these? I feel strongly that we should have the components that one uses to _build_ popovers separate from the popovers themselves, but I'm otherwise agreeable to suggestions.~ I went through the popovers directory and the majority of them belong to a single, existing component, so that's where they will live; shared popovers will be their own components.
